src/main/java/io/cdap/plugin/gcp/bigquery/sink/AbstractBigQuerySink.java

```diff
@@ -282,19 +340,13 @@ protected void validateSchema(Table table, Schema tableSchema, boolean allowSche
       return;
     }
 
-    if (getConfig().isTruncateTableSet()) {
-      //no validation required for schema if truncate table is set.
-      // BQ will overwrite the schema for normal tables when write disposition is WRITE_TRUNCATE
-      //note - If write to single partition is supported in future, schema validation will be necessary
-      return;
-    }
-
     FieldList bqFields = bqSchema.getFields();
     List<Schema.Field> outputSchemaFields = Objects.requireNonNull(tableSchema.getFields());
 
     List<String> missingBQFields = BigQueryUtil.getSchemaMinusBqFields(outputSchemaFields, bqFields);
 
-    if (allowSchemaRelaxation) {
+    if (allowSchemaRelaxation && !getConfig().isTruncateTableSet()) {
```

Comment:
Was this giving a runtime exception before?

Reply:
Nvm, the previous check was returning.

Reply:
This is just refactoring based on Albert's comment.